packages/next/src/export/index.ts

```diff
@@ -1,20 +1,20 @@
-import type { WorkerRenderOptsPartial } from './types'
+import type {
+  ExportAppResult,
+  ExportAppOptions,
+  ExportWorker,
+  WorkerRenderOptsPartial,
+} from './types'
+import type { PrerenderManifest } from '../build'
+import type { PagesManifest } from '../build/webpack/plugins/pages-manifest-plugin'
 
 import chalk from 'next/dist/compiled/chalk'
 import findUp from 'next/dist/compiled/find-up'
-import {
-  promises,
-  existsSync,
-  exists as existsOrig,
-  readFileSync,
-  writeFileSync,
-} from 'fs'
+import fs from 'fs/promises'
 
```

Comment:
Using sync fs calls is bad for performance, so I swapped everything over to use the promise API. This way multiple FS operations can be queued up and not stop-the-world to complete.